### PR TITLE
Content migration batch 12

### PR DESCRIPTION
**Pull request by mattletford**
Migrating content

https://developers.google.com/web/fundamentals/performance/get-started/wrapup-7
https://web.dev/performance-get-started-wrapup-7

https://developers.google.com/web/fundamentals/performance/http2
https://web.dev/performance-http2

https://developers.google.com/web/fundamentals/performance/optimizing-content-efficiency
https://web.dev/performance-optimizing-content-efficiency

https://developers.google.com/web/fundamentals/performance/optimizing-content-efficiency/client-hints
https://web.dev/optimizing-content-efficiency-client-hints

https://developers.google.com/web/fundamentals/performance/optimizing-content-efficiency/eliminate-downloads
https://web.dev/optimizing-content-efficiency-eliminate-downloads

@rachelandrew

### CODE REVIEW

File: src/site/content/en/blog/performance-get-started-wrapup-7/index.md [hunk 1]
```diff
@@ -0,0 +1,519 @@
...
+    <td>92%</td>
+  </tr>
+  <tr>
+    <td>singularity.jpg</td>
+    <td>jpg</td>
+    <td>158k</td>
+    <td>10k</td>
+    <td>94%</td>
+  </tr>
+</table>
```
Comment (malchata) [inline src/site/content/en/blog/performance-get-started-wrapup-7/index.md:277]:
```suggestion
</table>
</div>
```

File: src/site/content/en/blog/performance-get-started-wrapup-7/index.md [hunk 2]
```diff
@@ -0,0 +1,519 @@
...
+   by the physical size (width/height) reduction and image compression. The dimensions of the
+   hero images for the articles were reduced by one-third, from 900x500px to 600x333px.
+   Interestingly, that 33% reduction typically resulted in a 50% file size difference.
+
+3. Although compression on the large images at default settings (no special tweaking)
+   delivered both surprisingly low (3%) and
+   satisfyingly high (50%) results, the single biggest improvement factor across the board
+   was _reduction in jpg quality_. All jpgs were saved at 50% quality but still retained
+   their visual clarity, resulting in much smaller file sizes at no visual perception cost.
+   (We probably could have saved them at even lower qualities, but stopped at 50%.)
```
Comment (malchata) [inline src/site/content/en/blog/performance-get-started-wrapup-7/index.md:295]:
```suggestion
1. While some of the jpg images were slightly smaller when saved as gifs, their quality
   was visibly compromised. Thus, in each case, the "best" file type turned out to be the
   original file type, even if it was a bit larger than the smallest saved-as type.
2. Any size retention as a result of that visual inspection was completely compensated for
   by the physical size (width/height) reduction and image compression. The dimensions of the
   hero images for the articles were reduced by one-third, from 900x500px to 600x333px.
   Interestingly, that 33% reduction typically resulted in a 50% file size difference.
3. Although compression on the large images at default settings (no special tweaking)
   delivered both surprisingly low (3%) and
   satisfyingly high (50%) results, the single biggest improvement factor across the board
   was _reduction in jpg quality_. All jpgs were saved at 50% quality but still retained
   their visual clarity, resulting in much smaller file sizes at no visual perception cost.
   (We probably could have saved them at even lower qualities, but stopped at 50%.)
```

File: src/site/content/en/blog/performance-get-started-wrapup-7/index.md [hunk 3]
```diff
@@ -0,0 +1,519 @@
...
+## All Together Now
+
+Now that we've seen some speed improvements resulting from individual techniques, let's see
+what happens when we apply all the techniques in one version. For this test, we took these steps:
+
+- Minified the HTML, CSS, and JavaScript files as in test 1
+
+- Optimized the images as in test 2
+
+- Combined the CSS and JavaScript files, and the quotes and hero images as in test 3
```
Comment (malchata) [inline src/site/content/en/blog/performance-get-started-wrapup-7/index.md:428]:
```suggestion
- Minified the HTML, CSS, and JavaScript files as in test 1
- Optimized the images as in test 2
- Combined the CSS and JavaScript files, and the quotes and hero images as in test 3
```

File: src/site/content/en/blog/performance-get-started-wrapup-7/index.md [hunk 4]
```diff
@@ -0,0 +1,519 @@
...
+
+Here is the all-techniques version:
+[https://page-load-speed-demo.firebaseapp.com/pageload4.html](https://page-load-speed-demo.firebaseapp.com/pageload4.html).
+
+## Summary
+
+What can we take away from all these techniques, tests, and reports? Let's look at
+some basic numbers from the various test runs.
+
+<table>
```
Comment (malchata) [inline src/site/content/en/blog/performance-get-started-wrapup-7/index.md:456]:
```suggestion
<div class="table-wrapper scrollbar">
<table>
```

File: src/site/content/en/blog/performance-get-started-wrapup-7/index.md [hunk 5]
```diff
@@ -0,0 +1,519 @@
...
+    <td>1.3s</td>
+  </tr>
+  <tr>
+    <td>All</td>
+    <td>88</td>
+    <td>.47s, 1.1s</td>
+    <td>97</td>
+    <td>.96s</td>
+  </tr>
+</table>
```
Comment (malchata) [inline src/site/content/en/blog/performance-get-started-wrapup-7/index.md:499]:
```suggestion
</table>
</div>
```

File: src/site/content/en/blog/performance-get-started-wrapup-7/index.md [hunk 6]
```diff
@@ -0,0 +1,519 @@
...
+Finally, although the numbers for the various techniques tend to jump around a bit, it
+is satisfying to note that we achieved the overall fastest load and interactive times,
+as well as the highest percentile ranking among other tested pages, when we applied all
+the techniques at once.
+
+It's clear from these tests that the more improvement techniques you can bring to bear
+on a slow-loading page, the faster you can make it, and the better your users' experience
+will be.
+
+## Feedback {: #feedback }
```
Comment (malchata) [inline src/site/content/en/blog/performance-get-started-wrapup-7/index.md:519]:
```suggestion
```

File: src/site/content/en/blog/performance-http2/index.md [hunk 7]
```diff
@@ -0,0 +1,589 @@
...
+
+HTTP/2 does not modify the application semantics of HTTP in any way. All the
+core concepts, such as HTTP methods, status codes, URIs, and header fields,
+remain in place. Instead, HTTP/2 modifies how the data is formatted (framed) and
+transported between the client and server, both of which manage the entire
+process, and hides all the complexity from our applications within the new
+framing layer. As a result, all existing applications can be delivered without
+modification.
+
+## Why not HTTP/1.2?\_
```
Comment (malchata) [inline src/site/content/en/blog/performance-http2/index.md:41]:
```suggestion
## Why not HTTP/1.2?
```

File: src/site/content/en/blog/performance-http2/index.md [hunk 8]
```diff
@@ -0,0 +1,589 @@
...
+  – The full article by Ilya Grigorik
+- [“Setting up HTTP/2”](https://dassur.ma/things/h2setup/)
+  – How to set up HTTP/2 in different backends by Surma
+- [“HTTP/2 is here,
+  let’s optimize!”](https://docs.google.com/presentation/d/1r7QXGYOLCh4fcUq0jDdDwKJWNqWK1o4xMtYpKZCJYjM/edit#slide=id.p19)
+  – Presentation by Ilya Grigorik from Velocity 2015
+- [“Rules of Thumb for HTTP/2 Push”](https://docs.google.com/document/d/1K0NykTXBbbbTlv60t5MyJvXjqKGsCVNYHyLEXIxYMv0/edit)
+  – An analysis by Tom Bergan, Simon Pelchat and Michael Buettner on when and how to use push.
+
+## Feedback {: #feedback }
```
Comment (malchata) [inline src/site/content/en/blog/performance-http2/index.md:589]:
```suggestion
```

File: src/site/content/en/blog/performance-optimizing-content-efficiency-client-hints/index.md [hunk 9]
```diff
@@ -0,0 +1,724 @@
...
+```
+
+In this example, the `/image` URL is a PHP script followed by parameters
+rewritten by
+[mod_rewrite](https://httpd.apache.org/docs/current/mod/mod_rewrite.html). It
+takes an image filename and additional parameters to help a back-end script
+choose the best image in the given conditions.
+
+I sense _“But isn’t this just reimplementing `<picture>` and `srcset` on the
+back-end?”_ is your first question.
```
Comment (malchata) [inline src/site/content/en/blog/performance-optimizing-content-efficiency-client-hints/index.md:456]:
```suggestion
I sense _"But isn’t this just reimplementing `<picture>` and `srcset` on the
back-end?"_ is your first question.
```

File: src/site/content/en/blog/performance-optimizing-content-efficiency-client-hints/index.md [hunk 10]
```diff
@@ -0,0 +1,724 @@
+---
+title: Adapting to Users with Client Hints
+description: >
+  Client hints are a set of HTTP request headers we can use to change how we deliver page resources based on characteristics of a user's device and network connection. In this article, you'll learn all about client hints, how they work, and a few ideas on how you can use them to make your site faster for users.
+authors:
+  - jeremywagner
```
Comment (malchata) [inline src/site/content/en/blog/performance-optimizing-content-efficiency-client-hints/index.md:6]:
```suggestion
  - jlwagner
```

Comment (malchata) [reply src/site/content/en/blog/performance-optimizing-content-efficiency-client-hints/index.md:6]: I was the author for this one, and I changed my author slug to match my LDAP.

File: src/site/content/en/blog/performance-optimizing-content-efficiency-client-hints/index.md [hunk 11]
```diff
@@ -0,0 +1,724 @@
...
+  ([Slides](https://jlwagner.net/talks/take-a-client-hint))
+- [Delivering Fast and Light Applications with
+  `Save-Data`](/web/fundamentals/performance/optimizing-content-efficiency/save-data/)
+
+_Thank you to [Ilya Grigorik](https://twitter.com/igrigorik), [Eric
+Portis](https://twitter.com/etportis), [Jeff Posnick](https://twitter.com/jeffposnick), [Yoav
+Weiss](https://twitter.com/yoavweiss), and [Estelle Weyl](https://twitter.com/estellevw) for their
+valuable feedback and edits on this article._
+
+## Feedback {: #feedback .hide-from-toc }
```
Comment (malchata) [inline src/site/content/en/blog/performance-optimizing-content-efficiency-client-hints/index.md:724]:
```suggestion
```

File: src/site/content/en/blog/performance-optimizing-content-efficiency/index.md [hunk 12]
```diff
@@ -0,0 +1,73 @@
...
+updated: 2018-08-17
+---
+
+Our web applications continue to grow in their scope, ambition, and functionality -- that's a good thing. However, the relentless march toward a richer web is driving another trend: the amount of data downloaded by each application continues to increase at a steady pace. To deliver great performance we need to optimize delivery of each and every byte!
+
+What does a modern web application look like? [HTTP Archive](http://httparchive.org/) can help us answer this question. The project tracks how the web is built by periodically crawling the most popular sites (300,000+ from the Alexa Top 1M list) and recording and aggregating analytics on the number of resources, content types, and other metadata for each individual destination.
+
+{% Img src="image/C47gYyWYVMMhDmtYSLOWazuyePF2/JqsYuCtNUMTMm3KsDp5u.png", alt="HTTP Archive trends", width="597", height="217" %}
+
+<table class="">
```
Comment (malchata) [inline src/site/content/en/blog/performance-optimizing-content-efficiency/index.md:17]:
```suggestion
<div class="table-wrapper scrollbar">
<table class="">
```

File: src/site/content/en/blog/performance-optimizing-content-efficiency/index.md [hunk 13]
```diff
@@ -0,0 +1,73 @@
...
+  <td data-th="75%">308 KB</td>
+  <td data-th="90%">353 KB</td>
+</tr>
+<tr>
+  <td data-th="type"><strong>Total</strong></td>
+  <td data-th="50%"><strong>1054 KB</strong></td>
+  <td data-th="75%"><strong>1985 KB</strong></td>
+  <td data-th="90%"><strong>3486 KB</strong></td>
+</tr>
+</table>
```
Comment (malchata) [inline src/site/content/en/blog/performance-optimizing-content-efficiency/index.md:63]:
```suggestion
</table>
</div>
```

File: src/site/content/en/blog/performance-optimizing-content-efficiency/index.md [hunk 14]
```diff
@@ -0,0 +1,73 @@
...
+
+Unlike their desktop counterparts, web applications do not require a separate installation process: enter the URL and we are up and running -- that’s a key feature of the web. However, to make this happen **we often have to fetch dozens, and sometime hundreds, of various resources, all of which can add up to megabytes of data and must come together in hundreds of milliseconds to facilitate the instant web experience we are aiming for.**
+
+Achieving an instant web experience in light of these requirements is no small feat, which is why optimizing content efficiency is critical: eliminating unnecessary downloads, optimizing transfer encoding of each resource through various compression techniques, and leveraging caching whenever possible to eliminate redundant downloads.
+
+## Feedback {: #feedback }
```
Comment (malchata) [inline src/site/content/en/blog/performance-optimizing-content-efficiency/index.md:73]:
```suggestion
```

**Review by malchata — APPROVED** [review]
LGTM with some optional comments to address.

File: src/site/content/en/blog/performance-optimizing-content-efficiency-client-hints/index.md [hunk 15]
```diff
@@ -0,0 +1,722 @@
...
+
+**Extrinsic size:** the size of a media resource after CSS and other layout
+factors (such as `width` and `height` attributes) have been applied to it. Let’s
+say you have an `<img>` element that loads an image with a density-corrected
+intrinsic size of 320x240, but it also has CSS `width` and `height` properties
+with values of `256px` and `192px` applied to it, respectively. In this example,
+the _extrinsic size_ of that `<img>` element becomes 256x192.
+
+<figure>
+{% Img src="image/C47gYyWYVMMhDmtYSLOWazuyePF2/mYLOc0CZtVxXCNDoHcsK.svg", alt="An illustration of intrinsic size versus
```
Comment (rachelandrew) [inline src/site/content/en/blog/performance-optimizing-content-efficiency-client-hints/index.md:152]: I think the quotes in the alt are what the linter is complaining about.

**Review by rachelandrew — CHANGES_REQUESTED** [review]
The presubmit tests fail, I think it is due to the image I've left a comment about.

File: src/site/content/en/blog/performance-optimizing-content-efficiency-client-hints/index.md (hunk 15 above)
Comment (mattletford) [reply src/site/content/en/blog/performance-optimizing-content-efficiency-client-hints/index.md:152]: @rachelandrew Thank you, I've removed the quotes from the images

**Review by rachelandrew — APPROVED** [review]

**Pull request merged**